dags/ethereumetl_airflow/parse/parse_dataset_folder_logic.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import time
+
+from ethereumetl_airflow.common import get_list_of_files, read_json_file
+from ethereumetl_airflow.parse.parse_logic import parse, ref_regex
+from ethereumetl_airflow.parse.toposort import toposort_flatten
+
+
+def parse_dataset_folder(
+        bigquery_client,
+        dataset_folder,
+        ds,
+        source_project_id,
+        source_dataset_name,
+        destination_project_id,
+        sqls_folder,
+        parse_all_partitions,
+        time_func=time.time
+):
+    logging.info(f'Parsing dataset folder {dataset_folder}')
+    json_files = get_list_of_files(dataset_folder, '*.json')
+    logging.info(json_files)
+
+    topologically_sorted_json_files = topologically_sort_json_files(json_files)
+    logging.info(f'Topologically sorted json files: {topologically_sorted_json_files}')
+
+    for index, json_file in enumerate(topologically_sorted_json_files):
+        logging.info(f'Parsing json file {index} out of {len(topologically_sorted_json_files)}: {json_file}')
```

Comment:
Currently when someone adds a new table to a dateset, we will need to clear the dag for the table definitions to be parsed. Would be useful if we can skip the tables we don't need to re-parse since some datasets have a large number of table defs and this could become costly. This is especially true if we need to `parse_all_partitions` 

Maybe this is an opportunity to re-think how we merge PRs too and populate this? Maybe we could just the run the parse in  CI/CD so we don't need to clear tasks/dags on PR merge. Will also make the process of adding new tables much smoother. Especially in the cases where the same table defenition has been updated and this involves clearing the historical table or using load_all_partitions to correctly deploy this change. If this just runs on CI/CD , we wouldn't need to bother with making any changes to Composer.